src/common/transformations/src/transformations/low_precision/mark_dequantization_subgraph.cpp

```diff
@@ -62,6 +63,16 @@ ov::pass::MarkDequantizationSubgraph::MarkDequantizationSubgraph(const element::
         if (ov::op::util::is_on_constant_path(input)) {
             // disable ConstantFolding if dequantization subgraph is on constant data
             ov::disable_constant_folding(convert);
+            // It is also necessary to avoid precision conversion for constant nodes with input_precision
+            auto keep_const_precision = [&](Node* node) {
+                if (auto constant = ov::as_type<ov::opset10::Constant>(node)) {
+                    const auto& const_et = constant->get_element_type();
+                    if (std::find(precisions.begin(), precisions.end(), const_et) != precisions.end())
+                        ov::enable_keep_const_precision(convert->get_input_node_shared_ptr(0));
+                }
+            };
+            std::unordered_set<Node*> visited;
+            ov::op::util::visit_shape_path(input.get_node(), visited, keep_const_precision);
```

Comment:
Could you elaborate on this? Do we quantize ShapeOf sub-graphs now?

Reply:
let me redirect you question to @v-Golubev who is the author of the changes in this transformation.

Reply:
@jane-intel `visit_shape_path` is used here to traverse a constant subgraph which usually don't contain ShapeOf operations. So It's probably better to create a separate helper (smth like `visit_constant_path`) for that to avoid any confusion. If you agree I will do that in a separate PR.

Reply:
@jane-intel the new function is introduced in #20709. Could you please review? Thanks